java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java

```diff
@@ -0,0 +1,109 @@
+package com.zoolatech.lecture2.tasks._2;
+
+public class Robot {
+    private int w, h, n;
+    // direction = 0 - move right
+    // direction = 1 - move up
+    // direction = 2 - move left
+    // direction = 3 - move down
+    private int direction;
+
+    public Robot(int n) {
+        this.direction =0;
+        this.w = 0;
+        this.h = 0;
+        this.n = n-1;
+    }
+
+    public void move(int step){
+        if (this.direction==0){
+            this.w+=step;
```

Comment:
We add a positive value - why do we need to have an else block in the if-else block? It's enough to check only if a current coordinate is bigger than a room boundary. Similar for other directions.